src/Uno.UI/DataBinding/BindingPath.cs

```diff
@@ -430,8 +430,18 @@ public object DataContext
 				get => _dataContextWeakStorage?.Target;
 				set
 				{
-					if (!_disposed && DependencyObjectStore.AreDifferent(DataContext, value))
+					if (!_disposed)
 					{
+						// Historically, Uno was processing property changes using INPC. Since the inclusion of DependencyObject
+						// values changes are now filtered by DependencyProperty updates, making equality updates at this location
+						// detrimental to the use of INPC events processing.
+						// In case of an INPC, the bindings engine must reevaluate the path completely from the raising point, regardless
+						// of the reference being changed.
+						if(FeatureConfiguration.Binding.IgnoreINPCSameReferences && DependencyObjectStore.AreDifferent(DataContext, value))
```

Comment:
Not sure I understand what's happening but... is there a `!` missing here? Ie should it be returning early if (the feature flag is enabled and) the old and new values are _not_ different?

Reply:
Bad merge, will fix.